zephyr/Makefile

```diff
@@ -71,15 +70,10 @@ $(GENERIC_TARGETS): $(LIBMICROPYTHON)
 $(CLEAN_TARGETS):  clean
 
 $(GENERIC_TARGETS) $(KCONFIG_TARGETS) $(CLEAN_TARGETS):
-	$(RM) -f outdir/$(OUTDIR_PREFIX)/zephyr.lnk
 	$(MAKE) -f Makefile.zephyr BOARD=$(BOARD) CONF_FILE=$(CONF_FILE) $@
 
-$(LIBMICROPYTHON): $(Z_SYSGEN_H)
-build/genhdr/qstr.i.last: $(Z_SYSGEN_H)
-
-$(Z_SYSGEN_H):
-	rm -f $(LIBMICROPYTHON)
-	-$(MAKE) -f Makefile.zephyr BOARD=$(BOARD) CONF_FILE=$(CONF_FILE)
+$(LIBMICROPYTHON): | $(Z_EXPORTS)
```

Comment:
Make order-only deps are a bit weird, they don't work oftentime in DWIM manner. But here they did. The idea is that I'm telling that libmicropython should be be only after Z_EXPORTS is generated (which generates autoconf.h as a by-product, on which source files for libmicropython depend). But if i use a normal dependency, Z_EXPORTS (Makefile.exports) ends up inside .a file, which isn't cute. So, with order-only syntax I say: "depend on it, but not include in prerequisite list", and this time, it works as expected (it doesn't always ;-) ).

> it occurs to me we should say what we actually mean and make mpconfigport.h depend on sysgen.h

Well, sysgen.h is gone completely after https://github.com/micropython/micropython/commit/dbb2aea428fe2f9855784fc2fd04f7817b81360d , that's why refs to it are removed . And otherwise logic is per above and appear to work as expected.